src/platform/packages/private/kbn-index-editor/src/services/bulk_update_service.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { BulkRequest, BulkResponse } from '@elastic/elasticsearch/lib/api/types';
+import { LOOKUP_INDEX_UPDATE_ROUTE } from '@kbn/esql-types';
+import { groupBy, chunk } from 'lodash';
+import type { HttpStart } from '@kbn/core/public';
+import { ROW_PLACEHOLDER_PREFIX } from '../constants';
+import type { DeleteDocAction, DocUpdate } from '../types';
+import { isDocDelete, isDocUpdate } from '../utils';
+
+export type BulkUpdateOperations = Array<{ type: 'add-doc'; payload: DocUpdate } | DeleteDocAction>;
+
+// Update operations will be split in chunks of this size.
+export const BULK_UPDATE_CHUNK_SIZE = 500;
+
+/**
+ * Sends a bulk update request to an index.
+ * @param updates
+ */
+export async function bulkUpdate(
+  indexName: string,
+  updates: BulkUpdateOperations,
+  http: HttpStart
+): Promise<{
+  bulkOperations: Exclude<BulkRequest['operations'], undefined>;
+  bulkResponse: BulkResponse;
+}> {
+  const deletingDocIds: string[] = updates
+    .filter(isDocDelete)
+    .map((v) => v.payload.ids)
+    .flat();
+
+  const indexActions = updates.filter(isDocUpdate);
```

Comment:
So with indexActions we mean the updates right? Because also delete can also be an index action and I see we are differentiating them. Maybe a better variable name will make it more readable here?